spring-integration-sftp/src/test/java/org/springframework/integration/sftp/session/SftpSessionFactoryTests.java

```diff
@@ -343,4 +344,55 @@ protected SftpClient createSftpClient(ClientSession clientSession,
 		}
 	}
 
+	/*
+	 * Verify the socket is closed if authentication fails.
+	 */
+	@Test
+	public void testAuthFailSocketOpen() throws Exception {
+		try (SshServer server = SshServer.setUpDefaultServer()) {
+			server.setPort(0);
```

Comment:
I deliberately failed it with the `server.setPasswordAuthenticator((arg0, arg1, arg2) -> false);`.
There is no reason to loop for 10 times.